github_api_toolkit/__init__.py

```diff
@@ -155,3 +153,162 @@ def post(self, url: str, params: dict = {}, add_prefix: bool = True) -> requests
         if add_prefix:
             url = "https://api.github.com" + url
         return self.handle_response(requests.post(url=url, headers=self.headers, json=params))
+
+class github_graphql_interface():
+    """A class used to interact with the GitHub GraphQL API. Has a set range of functions.
+    """
+    
+    def __init__(self, token: str) -> None:
+        self.headers = { "Authorization": "token " + token }
+        self.api_url = "https://api.github.com/graphql"
+
+    def make_ql_request(self, query: str, params: dict) -> requests.Response:
+        """Makes a request to the GitHub GraphQL API.
+
+        Args:
+            query (str): The GraphQL query to be executed.
+            params (dict): A dictionary containing the variables for the query.
+
+        Returns:
+            requests.Response: The response from the API endpoint.
+        """
+
+        self.json = {
+            'query': query,
+            'variables': params
+        }
+
+        return requests.post(url=self.api_url, json=self.json, headers=self.headers)
+
+    def get_domain_email_by_user(self, username: str, org: str) -> list | tuple:
+        """Gets a GitHub user's verified domain email for a specific organization.
+
+        Args:
+            username (str): The GitHub username of the user.
+            org (str): The GitHub organization name.
+
+        Returns:
+            list | tuple: A list of verified domain emails for the user in the organization or a tuple containing an error message and status code.
+        """
+        
+        self.query = '''
+            query ($username: String!, $org: String!) {
+                user (login: $username) {
+                    login
+                    organizationVerifiedDomainEmails(login: $org)
+                }
+            }
+        '''
+
+        self.params = {
+            'username': username,
+            'org': org
+        }
+
+        response = self.make_ql_request(self.query, self.params)
+
+        if response.status_code == 200:
+            return response.json()["data"]["user"]["organizationVerifiedDomainEmails"]
+        else:
+            response_json = response.json()
+            return response_json["message"], response_json["status"]
```

Comment:
Is there a danger that the response may not include message? or status?
It may be safer with the following:

return response_json.get("message", "No message"), response_json.get("status", "Unknown status")